src/Build/Utilities/NuGetFrameworkWrapper.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.Reflection;
+using Microsoft.Build.Shared;
+
+namespace Microsoft.Build.Evaluation
+{
+    /// <summary>
+    /// Wraps the NuGet.Frameworks assembly, which is referenced by reflection.
+    /// </summary>
+    internal class NuGetFrameworkWrapper
+    {
+        /// <summary>
+        /// Resolve the location of the NuGet.Frameworks assembly
+        /// </summary>
+        private static Assembly GetNuGetAssembly()
```

Comment:
Should this be in a static ctor so it's called only once? Right now it'll be called for each exposed type, right?